### DISCUSSION TITLE
Code for setting order and location of EC shards?

### DISCUSSION THREAD

**Discussion by keirafadams** [minio/minio#15776] [Q&A] [answered]
Hi folks,

Generally mucking about, and maybe folks can short-circuit my search. Was looking for where in the PUT object path the data/parity distribution is set? i.e. which shards/chunks go to which disks/servers.

Thanks!

Ian

**Comment by klauspost** [comment, marked as answer]
Data and parity are placed on different disks for different objects. Mainly to distribute read load across all disks.

In practice only the "first" disk is selected randomly and data is stored in order on the following disks in a set.

Example metadata:
```
        "EcDist": [3,4,5,1,2],
        "EcIndex": 3,
        "EcM": 3,
        "EcN": 2,
```

3 data parity shards and 2 parity shards.

Data starts on disk 3. This means that for this object disk 3,4,5 contain data and 1+2 contain parity.

Code for new objects that determines distribution is here: https://github.com/minio/minio/blob/9e5853ecc01e62a0b1af1bd605ed11aa7e49df2e/cmd/storage-datatypes.go#L306

**Reply by keirafadams** [reply]
Excellent, was looking for the code that determines the distribution. Thanks!
